go-controller/pkg/ovn/controller/egress_services/egress_services_controller.go

```diff
@@ -141,31 +141,43 @@ func NewController(
 		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
 		"egressservices",
 	)
-	serviceInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+	_, err := serviceInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
 		AddFunc:    c.onServiceAdd,
 		UpdateFunc: c.onServiceUpdate,
 		DeleteFunc: c.onServiceDelete,
 	}))
+	if err != nil {
+		klog.Fatalf("Error adding event hander for service informer: %v", err)
```

Comment:
this should return error as it is eventually called by ```newDefaultNetworkControllerCommon``` which you added error handling for